### PR TITLE
docs: add debug build

### PR DESCRIPTION
**Pull request by BlackHole1**
#### Description of Change

add debug build instructions

Related discussion: https://github.com/electron/electron/pull/31950

cc: @deepak1556 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

**Comment by miniak** [comment]
Debug builds were removed on purpose here #21016

**Comment by deepak1556** [comment]
https://github.com/electron/electron/blob/main/docs/development/debugging.md might be a better home for this config addition.

**Comment by BlackHole1** [comment]
> Debug builds were removed on purpose here #21016

yes. However, we discussed this and felt it should be stated in the documentation. see: https://github.com/electron/electron/pull/31950#issuecomment-976398096

**Comment by BlackHole1** [comment]
@deepak1556 Please re-review

File: .gitignore [hunk 1]
```diff
@@ -53,4 +53,6 @@ ts-gen
 # Used to accelerate builds after sync
 patches/mtime-cache.json
 
-spec/fixtures/logo.png
\ No newline at end of file
+spec/fixtures/logo.png
+
+build/custom_args/*
```
Comment (deepak1556) [inline .gitignore:58]: Maybe this can be added to your local `.git/info/exclude` and not included in the tree.

**Review by deepak1556 — COMMENTED** [review]
LGTM with minor nit

**Review by jkleinsc — CHANGES_REQUESTED** [review]
I'm fine with this PR as long as the `.gitignore` changes are removed.

**Comment by BlackHole1** [comment]
@deepak1556 @jkleinsc done

**Review by miniak — APPROVED** [review]

File: docs/development/debugging.md [hunk 2]
```diff
@@ -35,6 +35,28 @@ base::debug::StackTrace().Print();
...
 This will allow you to observe call chains and identify potential issue areas.
 
+## Breakpoint Debugging
+
+> Note that this will increase the size of the build significantly, taking up around 50G of disk space
+
+Write the following file to `electron/.git/info/exclude/debug.gn`
+
+```gn
+import("../args/testing.gn")
```
Comment (deepak1556) [inline docs/development/debugging.md:45]: Always prefer absolute component paths `//electron/build/args/testing.gn`

Comment (BlackHole1) [reply docs/development/debugging.md]: done

**Review by deepak1556 — APPROVED** [review]

**Comment by BlackHole1** [comment]
@deepak1556 Can it be merged?

**Review by jkleinsc — APPROVED** [review]

**Comment by BlackHole1** [comment]
> One thing that I don't really grasp is how the developers and maintainers of Electron can work without proper debug support and component builds. I've tried developing with the testing configuration but when I started debugging I realized you have no access to local variables, members, etc., barely just a stack trace. Completely unusable for me, especially as I was digging into Chromium code too.

@kromain I'm as confused as you are...

@deepak1556 @codebytere Can you help us to answer this question?

**Comment by jviotti** [comment]
We also just stumbled into this and are having a very hard time doing any meaningful LLDB debugging with the current `testing` builds. Furthermore, adding the suggested GN arguments doesn't seem to work out of the box for macOS only:

```
FAILED: obj/electron/electron_framework_shared_library/Electron Framework obj/electron/electron_framework_shared_library/Electron Framework.TOC 
if [ ! -e "obj/electron/electron_framework_shared_library/Electron Framework" -o ! -e "obj/electron/electron_framework_shared_library/Electron Framework.TOC" ] || /Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool -l "obj/electron/electron_framework_shared_library/Electron Framework" | grep -q LC_REEXPORT_DYLIB ; then TOOL_VERSION=1654271340 ../../../../src/build/toolchain/apple/linker_driver.py -Wcrl,strippath,/Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip --deterministic ../../../../src/third_party/llvm-build/Release+Asserts/bin/clang++ -B /Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ -shared -Wl,-install_name,@rpath/Electron\ Framework.framework/Electron\ Framework -rpath @loader_path/Libraries -Wl,-force_load,obj/third_party/electron_node/deps/uv/libuv.a -Wl,-fatal_warnings -stdlib=libc++ -arch x86_64 -Werror -nostdlib++ -isysroot ../../../../.talaria/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -mmacosx-version-min=10.11.0 -Wl,-ObjC -L/Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/lib -F. -o "obj/electron/electron_framework_shared_library/Electron Framework" -Wl,-filelist,"obj/electron/electron_framework_shared_library/Electron Framework.rsp" -framework IOSurface -framework AVFoundation -framework Carbon -framework LocalAuthentication -framework QuartzCore -framework Quartz -framework Security -framework SecurityInterface -framework ServiceManagement -framework StoreKit -framework Squirrel -framework ReactiveObjC -framework Mantle -framework ApplicationServices -framework AppKit -framework CoreFoundation -framework IOKit -framework OpenDirectory -framework Foundation -framework CFNetwork -framework CoreServices -framework SystemConfiguration -framework CoreMedia -framework CoreGraphics -framework CoreText -framework Metal -framework OpenGL -framework Cocoa -framework Accelerate -framework AudioUnit -framework CoreVideo -framework AudioToolbox -framework CoreAudio -framework VideoToolbox -framework IOBluetooth -framework CoreMIDI -framework MediaAccessibility -framework DiskArbitration -framework ForceFeedback -framework GameController -framework CoreWLAN -framework CoreLocation -framework ImageCaptureCore -weak_framework Metal -weak_framework MetalKit -weak_framework MediaPlayer -weak_framework Vision  ./libffmpeg.dylib -lbsm -lpmenergy -lpmsample -lresolv -lsandbox -lcups -lm && { /Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool -l "obj/electron/electron_framework_shared_library/Electron Framework" | grep LC_ID_DYLIB -A 5; /Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/nm -gPp "obj/electron/electron_framework_shared_library/Electron Framework" | cut -f1-2 -d' ' | grep -v U$$; true; } > "obj/electron/electron_framework_shared_library/Electron Framework.TOC"; else TOOL_VERSION=1654271340 ../../../../src/build/toolchain/apple/linker_driver.py -Wcrl,strippath,/Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip --deterministic ../../../../src/third_party/llvm-build/Release+Asserts/bin/clang++ -B /Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ -shared -Wl,-install_name,@rpath/Electron\ Framework.framework/Electron\ Framework -rpath @loader_path/Libraries -Wl,-force_load,obj/third_party/electron_node/deps/uv/libuv.a -Wl,-fatal_warnings -stdlib=libc++ -arch x86_64 -Werror -nostdlib++ -isysroot ../../../../.talaria/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -mmacosx-version-min=10.11.0 -Wl,-ObjC -L/Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/lib -F. -o "obj/electron/electron_framework_shared_library/Electron Framework" -Wl,-filelist,"obj/electron/electron_framework_shared_library/Electron Framework.rsp" -framework IOSurface -framework AVFoundation -framework Carbon -framework LocalAuthentication -framework QuartzCore -framework Quartz -framework Security -framework SecurityInterface -framework ServiceManagement -framework StoreKit -framework Squirrel -framework ReactiveObjC -framework Mantle -framework ApplicationServices -framework AppKit -framework CoreFoundation -framework IOKit -framework OpenDirectory -framework Foundation -framework CFNetwork -framework CoreServices -framework SystemConfiguration -framework CoreMedia -framework CoreGraphics -framework CoreText -framework Metal -framework OpenGL -framework Cocoa -framework Accelerate -framework AudioUnit -framework CoreVideo -framework AudioToolbox -framework CoreAudio -framework VideoToolbox -framework IOBluetooth -framework CoreMIDI -framework MediaAccessibility -framework DiskArbitration -framework ForceFeedback -framework GameController -framework CoreWLAN -framework CoreLocation -framework ImageCaptureCore -weak_framework Metal -weak_framework MetalKit -weak_framework MediaPlayer -weak_framework Vision  ./libffmpeg.dylib -lbsm -lpmenergy -lpmsample -lresolv -lsandbox -lcups -lm && { /Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool -l "obj/electron/electron_framework_shared_library/Electron Framework" | grep LC_ID_DYLIB -A 5; /Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/nm -gPp "obj/electron/electron_framework_shared_library/Electron Framework" | cut -f1-2 -d' ' | grep -v U$$; true; } > "obj/electron/electron_framework_shared_library/Electron Framework.tmp" && if ! cmp -s "obj/electron/electron_framework_shared_library/Electron Framework.tmp" "obj/electron/electron_framework_shared_library/Electron Framework.TOC"; then mv "obj/electron/electron_framework_shared_library/Electron Framework.tmp" "obj/electron/electron_framework_shared_library/Electron Framework.TOC" ; fi; fi
Undefined symbols for architecture x86_64:
  "platform_util::GetViewForWindow(gfx::NativeWindow)", referenced from:
      BoundsOverlapWithAnyOpenPrompt(gfx::Rect const&, content::WebContents*) in libchrome.a(autofill_popup_view_utils.o)
  "platform_util::GetParent(gfx::NativeView)", referenced from:
      BoundsOverlapWithAnyOpenPrompt(gfx::Rect const&, content::WebContents*) in libchrome.a(autofill_popup_view_utils.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Traceback (most recent call last):
  File "../../../../src/build/toolchain/apple/linker_driver.py", line 305, in <module>
    Main(sys.argv)
  File "../../../../src/build/toolchain/apple/linker_driver.py", line 98, in Main
    subprocess.check_call(compiler_driver_args, env=env)
  File "/Users/distiller/project/.talaria/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../../../../src/third_party/llvm-build/Release+Asserts/bin/clang++', '-B', '/Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/', '-shared', '-Wl,-install_name,@rpath/Electron Framework.framework/Electron Framework', '-rpath', '@loader_path/Libraries', '-Wl,-force_load,obj/third_party/electron_node/deps/uv/libuv.a', '-Wl,-fatal_warnings', '-stdlib=libc++', '-arch', 'x86_64', '-Werror', '-nostdlib++', '-isysroot', '../../../../.talaria/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk', '-mmacosx-version-min=10.11.0', '-Wl,-ObjC', '-L/Users/distiller/project/.talaria/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/lib', '-F.', '-o', 'obj/electron/electron_framework_shared_library/Electron Framework', '-Wl,-filelist,obj/electron/electron_framework_shared_library/Electron Framework.rsp', '-framework', 'IOSurface', '-framework', 'AVFoundation', '-framework', 'Carbon', '-framework', 'LocalAuthentication', '-framework', 'QuartzCore', '-framework', 'Quartz', '-framework', 'Security', '-framework', 'SecurityInterface', '-framework', 'ServiceManagement', '-framework', 'StoreKit', '-framework', 'Squirrel', '-framework', 'ReactiveObjC', '-framework', 'Mantle', '-framework', 'ApplicationServices', '-framework', 'AppKit', '-framework', 'CoreFoundation', '-framework', 'IOKit', '-framework', 'OpenDirectory', '-framework', 'Foundation', '-framework', 'CFNetwork', '-framework', 'CoreServices', '-framework', 'SystemConfiguration', '-framework', 'CoreMedia', '-framework', 'CoreGraphics', '-framework', 'CoreText', '-framework', 'Metal', '-framework', 'OpenGL', '-framework', 'Cocoa', '-framework', 'Accelerate', '-framework', 'AudioUnit', '-framework', 'CoreVideo', '-framework', 'AudioToolbox', '-framework', 'CoreAudio', '-framework', 'VideoToolbox', '-framework', 'IOBluetooth', '-framework', 'CoreMIDI', '-framework', 'MediaAccessibility', '-framework', 'DiskArbitration', '-framework', 'ForceFeedback', '-framework', 'GameController', '-framework', 'CoreWLAN', '-framework', 'CoreLocation', '-framework', 'ImageCaptureCore', '-weak_framework', 'Metal', '-weak_framework', 'MetalKit', '-weak_framework', 'MediaPlayer', '-weak_framework', 'Vision', './libffmpeg.dylib', '-lbsm', '-lpmenergy', '-lpmsample', '-lresolv', '-lsandbox', '-lcups', '-lm']' returned non-zero exit status 1
ninja: build stopped: subcommand failed
```

**Pull request merged**